plugins/ident/geoip

```diff
@@ -341,29 +341,23 @@ sub init_my_country_code {
 
 sub set_country_code {
     my $self = shift;
-    my $remote_ip = $self->qp->connection->remote_ip;
-
-    my $code = $self->{_geoip_city}
-             ? $self->get_country_code_gc($remote_ip)
-             : $self->get_country_code($remote_ip);
-
-    return if ! $code;
+    my $ip   = $self->qp->connection->remote_ip;
+    my $code = $self->get_country_code($ip) or return;
     $self->qp->connection->notes('geoip_country', $code);
     return $code;
 }
 
 sub get_country_code {
     my $self = shift;
-    my $ip = shift || $self->qp->connection->remote_ip;
-    if ($self->{_geoip_city}) {
-        return $self->get_country_code_gc($ip);
-    };
-    return $self->{_geoip}->country_code_by_addr($ip);
+    my $ip   = shift || $self->qp->connection->remote_ip;
+    return $self->get_country_code_gc($ip)            if $self->{_geoip_city};
+    return $self->{_geoip}->country_code_by_addr($ip) if $self->{_geoip};
+    return undef;
 }
 
 sub get_country_code_gc {
     my $self = shift;
-    my $ip = shift || $self->qp->connection->remote_ip;
+    my $ip   = shift || $self->qp->connection->remote_ip;
     $self->{_geoip_record} = $self->{_geoip_city}->record_by_addr($ip)
```

Comment:
this code looks redundant to me, considering `get_country_code_gc()` has exactly one caller, which already sets $ip in exactly the same way.  Can we either not pass $ip, or not set it in this sub?

I know there's something to be said for code that accepts any input... but this seems a bit overboard to me.

My personal preference would be to require that the $ip is passed, and handle it as an error condition if it's not.